packages/components/multiselect/src/Multiselect.tsx

```diff
@@ -0,0 +1,278 @@
+
+export interface MultiselectProps
+  extends CommonProps,
+    Pick<
+      TextInputProps,
+      | 'isDisabled'
+      | 'isInvalid'
+      | 'isReadOnly'
+      | 'isRequired'
+      | 'id'
+      | 'defaultValue'
+    > {
+  /** Select Options */
+  children?: React.ReactNode;
+
+  /**
+   * Set a custom icon for the text input
+   */
+  startIcon?: React.ReactElement;
+
+  /**
+   * Placeholder shown before selecting any elements. Defaults to 'Select one or more items'
+   */
+  placeholder?: string;
+
+  /**
+   * current Selected items, to be shown on the trigger button
+   */
+  currentSelection?: Array<string>;
+
+  /**
+   * If this is set to `true` there will be a search input inside the drawer
+   * @default false
+   */
+  hasSearch?: boolean;
+
+  /**
+   * Function called whenever the search input value changes
+   */
+  onSearchValueChange?: (event: React.ChangeEvent) => void;
+
+  /**
+   * This is the value will be passed to the `placeholder` prop of the input.
+   * @default "Search"
+   */
+  searchPlaceholder?: string;
+
+  /**
+   * A message that will be shown when it is not possible to find any option that matches the input value
+   * @default "No matches"
+   */
+  noMatchesMessage?: string;
+
+  /**
+   * Use this prop to get a ref to the input element of the component
+   */
+  searchInputRef?: React.Ref<HTMLInputElement>;
+
+  /**
+   * Sets the list to show its loading state
+   * @default false
+   */
+  isLoading?: boolean;
+
+  /**
+   * Use this prop to get a ref to the toggle button of the component
+   */
+  toggleRef?: React.Ref<HTMLButtonElement>;
+
+  /**
+   * Props to pass to the Popover (Dropdown) component
+   */
+  popoverProps?: Partial<PopoverProps> & {
+    /**
+     * It sets the max-height, in pixels, of the list
+     * The default value is the height of 5 single line items
+     * @default 180
+     */
+    listMaxHeight?: number;
+
+    /**
+     * Use this prop to get a ref to the list of items of the component
+     */
+    listRef?: React.Ref<HTMLUListElement>;
+  };
+}
+
+function _Multiselect(props: MultiselectProps, ref: React.Ref<HTMLDivElement>) {
+  const {
+    id,
+    className,
+    startIcon,
+    placeholder = 'Select one or more Items',
+    currentSelection = [],
+    hasSearch = false,
+    defaultValue = '',
+    onSearchValueChange,
+    searchPlaceholder = 'Search',
+    searchInputRef,
+    noMatchesMessage = 'No matches found',
+    toggleRef,
+    isLoading = false,
+    testId = 'cf-multiselect',
+    popoverProps = {},
+    children,
+  } = props;
+
+  const { listMaxHeight = 180, listRef } = popoverProps;
+
+  const styles = getMultiselectStyles();
+
+  const [searchValue, setSearchValue] = useState(defaultValue);
+  const [isOpen, setIsOpen] = useState(false);
+
+  const internalSearchInputRef = useRef(null);
+
+  const handleSearchChange = (event) => {
+    setSearchValue(event.target.value);
+    onSearchValueChange(event);
```

Comment:
```suggestion
    onSearchValueChange?.(event);
```

You're saying that this needs to be provided but still, we might wanna make sure that we don't throw a plain error in this case.